lib/provisioning/Provisioner.js

```diff
@@ -91,62 +159,195 @@ Provisioner.prototype.isProvisionRequest = function(req) {
             req.url.match(/^\/_matrix\/provision\/listlinks/)
 };
 
-var parameterValidation = {
-    matrix_room_id :
-        {regex : /^!.*:.*$/, example : '!Abcdefg:example.com[:8080]'},
-    remote_room_channel :
-        {regex : /^([#+&]|(![A-Z0-9]{5}))[^\s:,]+$/, example : '#roomname'},
-    remote_room_server :
-        {regex : /^[a-z\.0-9:-]+$/, example : 'example.com or localhost'},
-};
+Provisioner.prototype._updateBridgingState = Promise.coroutine(
+    function*(roomId, userId, status, skey) {
+        let intent = this._ircBridge.getAppServiceBridge().getIntent();
+        try {
+            yield intent.client.sendStateEvent(roomId, 'm.room.bridging', {
+                user_id: userId,
+                status: status // pending, success, failure
+            }, skey);
+        }
+        catch (err) {
+            console.error(err);
+            throw new Error(`Could not update m.room.bridging state in this room`);
+        }
+    }
+);
+
+Provisioner.prototype._authoriseProvisioning = Promise.coroutine(
+    function*(server, userId, ircChannel, roomId, opNick, key) {
+        let ircDomain = server.domain;
+
+        let existing = this._getRequest(server, opNick);
+        if (existing) {
+            let from = existing.userId;
+            throw new Error(`Bridging request already sent to `+
+                            `${opNick} on ${server.domain} from ${from}`);
+        }
+
+        // (Matrix) Check power level of user
+        log.info(`Check power level of user`);
```

Comment:
Useless logging. Either add more info (which user, which room, etc) or remove this please.